core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -581,4 +582,9 @@ export class TopNavigationBarComponent implements OnInit, OnDestroy {
     return this.platformFeatureService.status
       .ShowFeedbackUpdatesInProfilePicDropdownMenu.isEnabled;
   }
+
+  shouldShowBackButton(): boolean {
+    let currentPath = this.urlService.getPathname();
+    return this.PAGES_WITH_BACK_STATE.some(path => currentPath.includes(path));
+  }
```

Comment:
Given that the output will not change in the same page, should we set it in a variable (`menuIconIsShown`) in the `ngOnInit` method and use that in the template?